apps/aetx/src/aetx.erl

```diff
@@ -154,19 +157,31 @@
     Tx :: tx().
 new(Callback, Tx) ->
     Type = Callback:type(),
-    #aetx{ type = Type, cb = Callback, tx = Tx }.
+    {Vsn, Fields} = Callback:serialize(Tx),
+    Template = Callback:serialization_template(Vsn),
+    Size = byte_size(aec_object_serialization:serialize(Type, Vsn, Template, Fields)),
+    #aetx{ type = Type, cb = Callback, size = Size, tx = Tx }.
 
 -spec fee(Tx :: tx()) -> Fee :: integer().
 fee(#aetx{ cb = CB, tx = Tx }) ->
     CB:fee(Tx).
 
+%% Each tx has gas computed based on the byte size of its serialized form. Some
+%% txs can add additional gas, e.g. contract txs that need the gas for contract
+%% execution.
 -spec gas(Tx :: tx()) -> Gas :: non_neg_integer().
-gas(#aetx{ cb = CB, tx = Tx }) ->
-    CB:gas(Tx).
-
--spec gas_price(Tx :: tx()) -> GasPrice :: non_neg_integer().
-gas_price(#aetx{ cb = CB, tx = Tx }) ->
-    CB:gas_price(Tx).
+gas(#aetx{ type = Type, cb = CB, size = Size, tx = Tx }) when
```

Comment:
Each TX should have a base cost as well?! It is not only the size... An oracle Tx is more complex than a spend so it should be more costly?!

Reply:
What base value do you suggest? I made it simple for now, there is no spec on how much more the oracle tx is more complex than the spend tx.

Reply:
No, those numbers don't exist yet... @ThomasArts is measuring... 

But the idea was that the cost of a TX was: TX cost + TX size * factor1 + State Tree Size * TTL * factor2

Though we don't have to fix all of that in this PR :-) But removing the `gas` callback from all TXs looks like a mistake?

Reply:
Just reinstate the 21000 or something a bit smaller and perhaps lower the 100 BYTE_COST a little bit and it is good enough here... And we can calibrate later.